drivers/sensor/bosch/bmi323/bmi323.c

```diff
@@ -1182,30 +1188,29 @@ static int bosch_bmi323_pm_resume(const struct device *dev)
 		return ret;
 	}
 
+#ifndef CONFIG_BMI323_BUS_I3C
```

Comment:
I understand why you would want to remove this... but It would really rather trust it doing a soft reset.

You probably removed this because in i3c, when do you the soft reset it, it stops working. That's because the soft reset also blows away it's DA (i dunno for sure, but I've seen this happen on other IMUs)
.. but I see it calls bus init right after that, so you probably should just re-init it's dynamic address with in there for it to work
(or there may be another 'type' of reset in that IMU that won't 'fully' reset it... but I dunno for sure)